### ISSUE TITLE
Signing of produced artifacts

### ISSUE THREAD

**Issue by lucasrod16** [defenseunicorns-labs/lula1#50]
Why do we need this?

We want to ensure the artifacts we publish, such as binaries and container images, have a unique signature to identify that they are the artifacts we built and published, and have not been tampered with.

We should aim to stay consistent with Zarf on this. They are using [cosign](https://github.com/sigstore/cosign), so we should look into using it as well.

**Comment by CloudBeard** [comment]
Do we want to setup release please as part of creating and signing the artifacts?

**Comment by brandtkeller** [comment]
Yeah.. I noticed this should likely have followed the releasing of artifacts. Will create an issue for that now.

**Comment by brandtkeller** [comment]
Blocked by #96

**Comment by CloudBeard** [comment]
@brandtkeller to do this we will need an AWS KMS key.

Cosign does support sign-blob keyless but its an [experimental feature](https://github.com/sigstore/cosign/blob/main/doc/cosign_sign-blob.md) 

Its worth tracking for donation to see when we can swap from KMS to keyless.

Ill throw up a draft PR to show what it can look like. We can publish the public key as an artifact too that way whenever the key is rotated in the backend we don't have to do anything it just gets published with the release it signs.

**Comment by CloudBeard** [comment]
https://github.com/defenseunicorns/lula/pull/353 

Still need to work with the security vanguard team to get the AWS creds then add that if we want to do it this way.

**Comment by CloudBeard** [comment]
Working with USV to get key/github secrets setup.

**Comment by CloudBeard** [comment]
Should be unblocked next week. Route forward has been determined.

**Comment by meganwolf0** [comment]
@CloudBeard new year new backlog review 🥳 - is this still blocked?
